lstgen/generators/go.py

```diff
@@ -0,0 +1,268 @@
+"""
+Go generator
+"""
+import ast
+from .. import (
+    parse_eval_stmt,
+    prepare_expr
+)
+from .. import (
+    EvalStmt,
+    IfStmt,
+    ThenStmt,
+    ElseStmt,
+    ExecuteStmt,
+)
+from .base import JavaLikeGenerator
+
+class GoGenerator(JavaLikeGenerator):
+    """ Java Generator """
+    bd_class = 'decimal.Decimal'
+    bd_class_constructor = 'decimal.NewFromInt'
+    list_const_parens = ('{', '}')
+    stmt_separator = ''
+    allow_constants = False
+
+    def __init__(self, parser, outfile, class_name=None, indent=None, package_name='default'):
+        super(GoGenerator, self).__init__(parser, outfile, class_name, indent)
+        self.package_name = package_name
+
+    def generate(self):
+        wr = self.writer
+        self._write_comment("This file is automatically generated by LstGen, do not edit!", False)
+        wr.writeln('package {}'.format(self.package_name or 'tax'))
+        wr.nl()
+        wr.writeln('import ( "github.com/shopspring/decimal" )')
```

Comment:
Maybe it's a good idea to make that BD package configurable via CLI too?

Reply:
Any replacement Decimal package would require 100% API compatibility, or specific mappings which would require a major redesign of the implementation, so I would not implement this.

Reply:
This could be problematic because not all packages that process the BD have the same interfaces. This means that you have to restrict yourself to just one.

Reply:
I agree that it would be nice to have, it just exceeds the amount of work I'm willing to put in.